src/rpc/masternode.cpp

```diff
@@ -413,46 +413,40 @@ UniValue startmasternode(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_MISC_ERROR, "startmasternode is not supported when deterministic masternode list is active (DIP3)");
     }
 
-    CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = GetWalletForJSONRPCRequest(request);
 
     if (!EnsureWalletIsAvailable(pwallet, request.fHelp))
         return NullUniValue;
 
     std::string strCommand;
-    if (request.params.size() >= 1) {
+    if (!request.params.empty()) {
         strCommand = request.params[0].get_str();
-
-        // Backwards compatibility with legacy 'masternode' super-command forwarder
-        if (strCommand == "start") strCommand = "local";
-        if (strCommand == "start-alias") strCommand = "alias";
-        if (strCommand == "start-all") strCommand = "all";
-        if (strCommand == "start-many") strCommand = "many";
-        if (strCommand == "start-missing") strCommand = "missing";
-        if (strCommand == "start-disabled") strCommand = "disabled";
     }
 
+    if (strCommand == "local")
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Local start is deprecated. Start your masternode from the controller wallet instead.");
+    if (strCommand == "many")
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Many set is deprecated. Use either 'all', 'missing', or 'disabled'.");
+
     if (request.fHelp || request.params.size() < 2 || request.params.size() > 4 ||
-        (request.params.size() == 2 && (strCommand != "local" && strCommand != "all" && strCommand != "many" && strCommand != "missing" && strCommand != "disabled")) ||
-        ( (request.params.size() == 3 || request.params.size() == 4) && strCommand != "alias"))
+        (strCommand == "alias" && request.params.size() < 3))
```

Comment:
Improve this if:
if the strCommand is "alias" you must use either 3 or 4 params
if the strCommand is valid but not "alias" you must use 2 params
if the strCommand is not valid you should throw error anyways

Reply:
> if the strCommand is "alias" you must use either 3 or 4 params

"alias" requires *at least* 3 arguments, this is specifically covered in this conditional check.


> if the strCommand is valid but not "alias" you must use 2 params

This is no longer the case, as shown in the changes to the functional tests that use named arguments with the "all" command whilst reloading the `masternode.conf` file ([here](https://github.com/PIVX-Project/PIVX/pull/2833/files#diff-4a04bc0b355c780033960e8c261ee9b6d3c452897e1dcd88a15d272512266c76R1038-R1043) and [here](https://github.com/PIVX-Project/PIVX/pull/2833/files#diff-e09c91077bce1ecdcd2a58fa99956e9b9f9f91ef0238ee1ce22b1d01bc828c04R86)).


> if the strCommand is not valid you should throw error anyways

This does that, and is tested in the functional tests ([here](https://github.com/PIVX-Project/PIVX/pull/2833/files#diff-000f98a6301bea2b42904854c3ad8b5c9fb15857fefddecb5f0d73bfc7da1c4fR98)) by returning a `RPC_INVALID_PARAMETER` error code. This is done towards the end of the RPC function call and not at the beginning intentionally ([here](https://github.com/PIVX-Project/PIVX/pull/2833/files#diff-244e87df3ed336aadc56ec7c8fc521954751f3c3f66b3eb3832bcbcc87996c51R540)).

Reply:
I moved to this approach for a few reasons:
1) It allows greater code coverage and testability.
2) A user's natural instinct upon getting an `RPC_INVALID_PARAMETER` error (**SHOULD**) be to then check the `help` output.
3) This new behavior is much much more inline with the rest of our RPC commands, and greatly simplifies the logic checks being done for when to show the actual help output.